pkg/segment/reader/record/rrcreader.go

```diff
@@ -181,6 +181,24 @@ func GetJsonFromAllRrc(allrrc []*utils.RecordResultContainer, esResponse bool, q
 
 	sort.Strings(colsSlice)
 	log.Infof("qid=%d, GetJsonFromAllRrc: Got %v raw records from files in %+v", qid, len(allRecords), time.Since(sTime))
+
+	if aggs != nil {
+		for post := aggs; post != nil; post = post.Next {
+			if post.TransactionArguments != nil {
+				recs := make(map[string]map[string]interface{}, 0)
+				recs["0"] = make(map[string]interface{})
+				recs["0"]["records"] = allRecords
+				recs["0"]["columns"] = colsSlice
+				nodeRes := &structs.NodeResult{}
+				agg.PostQueryBucketCleaning(nodeRes, aggs, recs, finalCols)
+				allRecords = recs["0"]["records"].([]map[string]interface{})
+				colsSlice = recs["0"]["columns"].([]string)
+				break
+			}
+		}
+
+	}
```

Comment:
There's some similar logic where we call `PostQueryBucketCleaning()` above this on line 112. If you're able to reuse that logic I think that would be better so the code doesn't get too bloated if we add more commands that need to do something like this.